src/page-modules/departures/server/journey-planner/index.ts

```diff
@@ -166,6 +166,21 @@ export function createJourneyApi(
         throw validated.error;
       }
 
+      validated.data.quays.sort((a, b) => {
+        // Place quays with no departures at the end
+        if (a.departures.length === 0) return 1;
+        if (b.departures.length === 0) return -1;
+
+        if (!a.publicCode) return 1;
+        if (!b.publicCode) return -1;
+
+        if (parseInt(a.publicCode) && parseInt(b.publicCode)) {
+          return parseInt(a.publicCode) - parseInt(b.publicCode);
+        }
```

Comment:
Something like

```suggestion
        if (a.departures.length === 0) return 1;
        if (b.departures.length === 0) return -1;
        
        const publicA = parseInt(a.publicCode, 10);
        const publicB = parseInt(b.publicCode, 10);
        
        if (Number.isNaN(publicB) || Number.isNaN(publica)) {
          return a.publicCode.localeCompare(b.publicCode);
        }

        return publicA - publicB;
```

There is something odd with the `if (parseInt(a.publicCode) && parseInt(b.publicCode))`, because if it is not parsed it is NaN so it will be if NaN && NaN which is technically false, but odd. But also if public code is 0 this will be false. Again, probably not any real world consequences in this specific case, but the check it self is kind of odd, so I wouldn't say it is a good check to have of principle.